aws/internal/service/ec2/waiter/status.go

```diff
@@ -206,6 +207,43 @@ func SecurityGroupStatus(conn *ec2.EC2, id string) resource.StateRefreshFunc {
 	}
 }
 
+const (
+	vpcPeeringConnectionStatusNotFound = "NotFound"
+	vpcPeeringConnectionStatusUnknown  = "Unknown"
+)
+
+// VpcPeeringConnectionStatus fetches the VPC peering connection and its status
+func VpcPeeringConnectionStatus(conn *ec2.EC2, vpcPeeringConnectionID string) resource.StateRefreshFunc {
```

Comment:
Can refactor existing VPC Peering Connection waiters to use this after merge.

Reply:
Opened https://github.com/terraform-providers/terraform-provider-aws/issues/15908 to track this.